src/debug-generator/debug-generator.c

```diff
@@ -136,9 +144,10 @@ static int generate_wants_symlinks(void) {
 }
 
 static void install_debug_shell_dropin(const char *dir) {
+        const char *tty = arg_debug_tty ?: arg_default_debug_tty;
         int r;
 
-        if (streq(arg_debug_shell, skip_dev_prefix(DEBUGTTY)))
+        if (!tty || streq(tty, skip_dev_prefix(DEBUGTTY)))
```

Comment:
maybe switch to path_equal() here. After all there are /dev/serial/by-id/ symlinks these days, and it should be OK to specify them, and in that we should properly compare htem.